projects/vdk-plugins/vdk-meta-jobs/src/vdk/plugin/meta_jobs/meta_dag.py

```diff
@@ -63,6 +63,10 @@ def execute_dag(self):
         while self._topological_sorter.is_active():
             for node in self._topological_sorter.get_ready():
                 self._start_job(node)
+                if len(self._job_executor.get_currently_running_jobs()) >= self._max_concurrent_running_jobs:
```

Comment:
The root cause was that as soon as the limit is hit, the method "_start_delayed_jobs()" is trying to be executed but there is no space left for the delayed jobs and that's why I decided that this is a good moment to finalise the current running jobs as soon as we hit the limit and be able to start the delayed ones.

Reply:
Actually just realized what I did there... Now I moved the check out of the loop.

Reply:
Yep, it makes sense that this would not fix the bug. It would indeed loop forever due to the number of delayed jobs exceeding the concurrent limit. I will change my approach to fix that.

Reply:
> loop forever due to the number of delayed jobs exceeding the concurrent limit.

I am not sure I see it. That does not seem to be the reason I see or I do not understand you correctly. 
Please, it's really important to read the code correctly and understand what the code does when making changes.

Reply:
By this I meant that if there are more than N delayed jobs (let's say N = limit), then when we go into "_start_delayed_jobs()" we would be forever stuck in the loop there, delaying the already delayed jobs because the running jobs hit the limit and there is no room for these delayed ones.

Reply:
It no longer hangs with the latest changes.